### PR TITLE
334 monsoon wang charles review [WIP]

### PR DESCRIPTION
**Pull request by doutriaux1**
do not merge yet

### CODE REVIEW

File: src/python/devel/monsoon_wang/graphics/bar_chart_monsoon_precip_index.py [hunk 1]
```diff
@@ -0,0 +1,412 @@
...
+   pth, fnm = os.path.split(args.parameters)
+   if pth != "":
+      sys.path.append(pth)
+   if fnm.lower()[-3:] == ".py":
+      fnm = fnm[:-3]
+      ext = ".py"
+   else:
+      ext = ""
+   # We need to make sure there is no "dot" in filename or import will fail
+   if fnm.find(".") > -1:
```
Comment (doutriaux1) [inline src/python/devel/monsoon_wang/graphics/bar_chart_monsoon_precip_index.py]: @zshaheen ???

**Comment by doutriaux1** [comment]
@gleckler1 c an you send me an example on how to run this? Thx!

**Comment by gleckler1** [comment]
@doutriaux1 Check out comment on line 17 of your code: 
https://github.com/PCMDI/pcmdi_metrics/blob/334_monsoon_wang_charles_review/src/python/devel/monsoon_wang/scripts/mpi_compute.py

**Comment by doutriaux1** [comment]
thx @gleckler1

File: src/python/devel/monsoon_wang/graphics/bar_chart_monsoon_precip_index.py [hunk 2]
```diff
@@ -0,0 +1,412 @@
...
+      title = parameters.title
+   # -- yaxis_label
+   if hasattr(parameters, 'yaxis_label') and not args.yaxis_label:
+      yaxis_label = parameters.yaxis_label
+   # -- highlights
+   if hasattr(parameters, 'highlights') and not args.highlights:
+      highlights = parameters.highlights
+
+
+print '-----------------------------'
```
Comment (zshaheen) [inline src/python/devel/monsoon_wang/graphics/bar_chart_monsoon_precip_index.py]: @doutriaux1 No, there's no verbosity arg. I'll add that into my fix for today.

File: src/python/devel/monsoon_wang/graphics/bar_chart_monsoon_precip_index.py (hunk 1 above)
Comment (zshaheen) [inline src/python/devel/monsoon_wang/graphics/bar_chart_monsoon_precip_index.py]: @doutriaux1 Yes, the parser already does this.

File: src/python/devel/monsoon_wang/lib/monsoon_precip_index_fncs.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
+import MV2
+#  SEASONAL RANGE - USING ANNUAL CYCLE CLIMATOLGIES 0=Jan, 11=Dec
+def mpd(d):
+ mjjas = (d[4] + d[5] + d[6] + d[7] + d[8])/5.
```
Comment (gleckler1) [inline src/python/devel/monsoon_wang/lib/monsoon_precip_index_fncs.py]: Yes

Comment (gleckler1) [inline src/python/devel/monsoon_wang/lib/monsoon_precip_index_fncs.py]: All clim statistics in the PMP driver expect this too.

File: src/python/devel/monsoon_wang/lib/monsoon_precip_index_fncs.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+ mpi = MV2.divide(annrange,ann)
+
+ return annrange, mpi
+
+def mpi_skill_scores(annrange_mod_dom,annrange_obs_dom,thr):
+   mt = MV2.where(MV2.greater(annrange_mod_dom,thr),1.,0.)
+   ot = MV2.where(MV2.greater(annrange_obs_dom,thr),1.,0.)
+
+   both = MV2.add(mt,ot) # HIT WILL MEAN 2, OTHERWISE 0 or 1
```
Comment (gleckler1) [inline src/python/devel/monsoon_wang/lib/monsoon_precip_index_fncs.py]: @doutriaux1 I thought it was better practice to use MV than + - / *

File: src/python/devel/monsoon_wang/graphics/bar_chart_monsoon_precip_index.py [hunk 5]
```diff
@@ -0,0 +1,412 @@
...
+                      type = str,
+                      dest = 'variable',
+                      default = '',
+                      help = "(Case Insensitive)")
+P.add_argument("-s", "--stat",
+                      type = str,
+                      default = 'rms',
+                      help = "Statistic:\n"
+                             "- Available options: bias, cor, rms")
+P.add_argument("-seas", "--season",
```
Comment (doutriaux1) [inline src/python/devel/monsoon_wang/graphics/bar_chart_monsoon_precip_index.py]: one dash should only be ONE letter, multiple letters with two dashes

**Comment by doutriaux1** [comment]
not ready at all!!!

**Comment by doutriaux1** [comment]
@gleckler1 this branch computation part is ready for you to give it a test drive

Things I need to do:

* [ ] Add a test case
* [ ] Make code use genutil.StringConstructors
* [ ] Cleanup graphics code
* [ ] Port graphics to VCS (very long term)

Things you need to do:

* [ ] Make sure this branch produces correct result
* [ ] Cleanup arguments so that they are consistent accross drivers (at least with pcmdi_metrics_driver)

**Comment by gleckler1** [comment]
Outstanding CD ... may the world improve for you (and the rest of us)!

**Comment by gleckler1** [comment]
@CD  Is GH pull of 334_monsoon_wang_charles_review they way for me to install this and test it out?

**Comment by doutriaux1** [comment]
yep, please pull and checkout this branch then
```
python setup.py install --enable-devel
```

Then 

```
mpi_compute -h
```

**Comment by durack1** [comment]
@gleckler1 FYI your "@ cd" above pinged the user found at https://github.com/cd

**Comment by gleckler1** [comment]
@doutriaux1  Followed instructions but something not right.   I expected refactored code to be in:
https://github.com/PCMDI/pcmdi_metrics/tree/334_monsoon_wang_charles_review/src/python**/devel**/monsoon_wang

not

https://github.com/PCMDI/pcmdi_metrics/tree/334_monsoon_wang_charles_review/src/python/monsoon_wang

This probably explains why:
bash-4.1$ python setup.py install --enable-devel
demo files
Adding experimental packages
running install
running build
running build_py
copying src/python/pcmdi/std_xy.py -> build/lib/pcmdi_metrics/pcmdi
copying src/python/pcmdi/mean_climate_metrics_calculations.py -> build/lib/pcmdi_metrics/pcmdi
copying src/python/__init__.py -> build/lib/pcmdi_metrics
.
.
.
error: package directory 'src/python/devel/monsoon_wang/lib' does not exist

**Comment by doutriaux1** [comment]
@gleckler1 yes i moved it into the main it's not in devel anymore since it's for merging in master. I had left stuff in devel because I think it was the graphics. I'll clean it so that --devel works again.

**Comment by gleckler1** [comment]
@doutriaux1 OKBYME.   I presume I should still do a test run before we merge.  To do that, should I use  python setup.py install 
instead of python setup.py install --enable-devel  ?

**Pull request merged**